src/c++/perf_analyzer/client_backend/triton_c_api/triton_loader.cc

```diff
@@ -894,14 +894,18 @@ TritonLoader::Infer(
     const std::vector<const tc::InferRequestedOutput*>& outputs,
     InferResult** result)
 {
+  if (!ServerIsReady() || !ModelIsLoaded()) {
+    return Error("Server is not ready and/or requested model is not loaded");
+  }
   TRITONSERVER_ResponseAllocator* allocator = nullptr;
   TRITONSERVER_InferenceRequest* irequest = nullptr;
-  Timer().Reset();
-  Timer().CaptureTimestamp(tc::RequestTimers::Kind::REQUEST_START);
+  tc::RequestTimers timer;
```

Comment:
Why not re-use the class member timer instead of creating a new one each time?

Reply:
Since `TritonLoader` is a singleton, there's only one instance of `timer` during its lifetime. If we wanted to enable `RequestRateManager` which uses multiple instances of the timer, we would need a new timer every `Infer` call